internal/plugin/connectors/tcp/mssql/connector_test.go

```diff
@@ -49,6 +51,76 @@ func TestHappyPath(t *testing.T) {
 
 	assert.Equal(t, expectedBackendConn, actualBackendConn)
 }
+
+func TestWritePreloginSucceeds(t *testing.T) {
+	logger := logmock.NewLogger()
+	clientConn := mock.NewNetConn(nil)
+	expectedBackendConn := mock.NewNetConn(nil)
+	creds := connector.CredentialValuesByID{
+		"credName": []byte("secret"),
+	}
+
+	// The fields we should send to the user
+	expectedPreloginResponse := map[uint8][]byte{
+		mssql.PreloginVERSION:    {0, 0, 0, 0, 0, 0},
+		mssql.PreloginENCRYPTION: {mssql.EncryptNotSup},
+		mssql.PreloginINSTOPT:    {0},
+		mssql.PreloginTHREADID:   {0, 0, 0, 0},
+		mssql.PreloginMARS:       {0}, // MARS disabled
+	}
+
+	// The fields we should get back from inside the mssql.Connect method
+	mockPreloginResponse := map[uint8][]byte{
+		mssql.PreloginVERSION:    {0, 0, 0, 0, 0, 0},
+		mssql.PreloginENCRYPTION: {mssql.EncryptOn},
+		mssql.PreloginINSTOPT:    {0},
+		mssql.PreloginTHREADID:   {0, 0, 0, 0},
+		mssql.PreloginMARS:       {0}, // MARS disabled
+	}
+
+	// Perform the actual write operation and
+	// assert that we are writing the correct fields back to the client
+	writePreloginWithAssertion := func(
+		tdsBuffer io.ReadWriteCloser,
+		fields map[uint8][]byte,
+		pktType uint8) error {
+			_ = mssql.WritePreloginWithPacketType(tdsBuffer, fields, pktType)
```

Comment:
We don't want any references to `mssql` in any of the UTs.

Also, you should rarely (never?) need assertions _inside_ of your dependency doubles. 

I believe for the test that you're trying to do here all you need to do in this double is write your `fields` into the `tdsBuffer` (which isn't really a `tdsBuffer`, it's just a `ReadWriteCloser`).  To make that last parenthetical point clear, you might want to name it `fakeTdsBuffer`.  We can completely ignore `pktType` as that is only needed by a real `tdsBuffer`.

Now you'll have a double that does nothing but write the bytes you give it  into the buffer you give it.  In the test, you'll assert on a buffer mock that it received a `Write` call with the parameter you expect.  Or on a buffer double that the you actual  bytes you expect were written to it.  I prefer the latter approach.